presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsCalculator.java

```diff
@@ -176,6 +182,7 @@ private Map<PlanCanonicalizationStrategy, PlanNodeWithHash> getPlanNodeHashes(Pl
     private PlanNodeStatsEstimate getStatistics(PlanNode planNode, Session session, Lookup lookup, PlanNodeStatsEstimate delegateStats)
     {
         if (!useHistoryBasedPlanStatisticsEnabled(session)) {
+            LOG.info("History based statistics are disabled for this session");
```

Comment:
This can create too much logging.

Reply:
@jaystarshot After removing this, do you still think too much logging will occur if a user is looking to debug why HBO was not used

Reply:
If thats your goal, why not add runtime metrics and publish them through the event listener?

Reply:
Let's collect the information and record the reasons in an external table using the event listener so as to make it easy to use.

In addition to why HBO is not available for a whole query, sometimes HBO is available for some of the nodes but not for others, you will also need to record why a specific plan node lacks HBO statistics.